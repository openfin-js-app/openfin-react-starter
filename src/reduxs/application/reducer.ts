import { handleActions, Action } from 'redux-actions';
import { Docking, System, Window, Event } from '@albertli90/redux-openfin';

import {
    IApplicationNewSnackbarOption,
    IApplicationSetSnackbarStatusOption,
    ISnackBarMsg, IApplicationState,
} from './types';

import {
    APPLICATION_READY,
    APPLICATION_DRAWER_TOGGLE,
    APPLICATION_NEW_SNACKBAR,
    APPLICATION_SET_SNACKBAR_STATUS,
    APPLICATION_PROCESS_SNACKBAR_QUEUE,
    APPLICATION_LAUNCH_BAR_TOGGLE_COLLAPSE,
    APPLICATION_NETWORK_ONLINE,
    APPLICATION_NETWORK_OFFLINE,
} from './actions';

export const defaultState:Partial<IApplicationState>={
    offline:false,
    username:'',
    computerName:'',
    machineId:null,
    deviceUserId:null,
    loading:true,
    docked:false,
    winTop:0,
    winLeft:0,
    winWidth:0,
    winHeight:0,
    drawerOpen:true,
    launchBarCollapse:false,
    snackBarOpen:false,
    snackBarMsgInfo:{},
    snackBarMsgQueue:[],
    openfinVersion:'n/a',
    openfinHostSpec:{},
    windowsState:'normal',
};

export default (parentWindowState?:Partial<IApplicationState>)=>{
    let initState:Partial<IApplicationState>;

    if (parentWindowState){
        initState ={
            ...parentWindowState,
            docked:false,
            winTop:0,
            winLeft:0,
            winWidth:0,
            winHeight:0,
            snackBarMsgInfo:{},
            snackBarMsgQueue:[],
            openfinHostSpec:{
                ...parentWindowState.openfinHostSpec,
            },
            windowsState:'normal',
        }

    }else{
        initState = defaultState;
    }

    return handleActions({
        [System.actions.GET_MACHINE_ID_RES]:(state,action)=>{
            const {id} = action.payload as any;
            return {
                ...state,
                machineId:id
            };
        },
        [System.actions.GET_DEVICE_USER_ID_RES]:(state,action)=>{
            const {id} = action.payload as any;
            return {
                ...state,
                deviceUserId:id
            };
        },
        [System.actions.GET_ENVIRONMENT_VARIABLE_RES]:(state,action)=>{
            const {env,value} = action.payload as any;
            if (env.toLowerCase()==='username'){
                return {
                    ...state,
                    username:value,
                }
            }else if ((env.toLowerCase() === 'computername' || env.toLowerCase() === 'hostname') && value){
                return {
                    ...state,
                    computerName:value,
                }
            }else{
                return state;
            }
        },
        [System.actions.GET_VERSION_RES]:(state,action)=>{
            const {version} = action.payload as any;
            return {
                ...state,
                openfinVersion:version,
            };

        },
        [System.actions.GET_HOST_SPECS_RES]:(state,action)=>{
            const openfinHostSpec = action.payload as any;
            return {
                ...state,
                openfinHostSpec
            };

        },
        [Window.actions.GET_STATE_RES]:(state,action)=>{
            const payload = action.payload as any;
            return {
                ...state,
                windowsState:payload.state,
            };

        },
        [Window.actions.GET_BOUNDS_RES]:(state,action)=>{
            const payload = action.payload as any;
            return {
                ...state,
                winTop:payload.top,
                winLeft:payload.left,
                winWidth:payload.width,
                winHeight:payload.height,
            };

        },
        [Event.actionDicts.windowEventDictByName['bounds-changing'].type]:(state,action)=>{
            const payload = action.payload as any;
            return {
                ...state,
                winTop:payload.top,
                winLeft:payload.left,
                winWidth:payload.width,
                winHeight:payload.height,
            };

        },
        [Event.actionDicts.windowEventDictByName['group-changed'].type]:(state,action)=>{
            const {
                sourceWindowName, targetWindowName, memeberOf, reason
            } = action.payload;

            if (reason === Docking.types.GroupEventReason.JOIN && sourceWindowName === window.name){
                return {
                    ...state,
                    docked:true,
                }
            }else if(reason === Docking.types.GroupEventReason.LEAVE && sourceWindowName === window.name){
                return {
                    ...state,
                    docked:false,
                }
            }

            return state;
        },
        [APPLICATION_READY]:(state,action)=>({
            ...state,
            loading:false,
        }),
        [APPLICATION_DRAWER_TOGGLE]:(state,action)=>({
            ...state,
            drawerOpen:!state.drawerOpen
        }),
        [APPLICATION_NEW_SNACKBAR]:(state,action)=>{
            const option:IApplicationNewSnackbarOption = action.payload as IApplicationNewSnackbarOption;
            const newMsgQueue = state.snackBarMsgQueue.concat([{
                message:option.message,
                key: new Date().getTime(),
                variant:option.variant,
            }]);
            return{
                ...state,
                snackBarMsgQueue:newMsgQueue,
            }
        },
        [APPLICATION_PROCESS_SNACKBAR_QUEUE]:(state,action)=>{
            if (state.snackBarMsgQueue.length > 0){
                const newMsg = state.snackBarMsgQueue[0];
                const newMsgQueue = state.snackBarMsgQueue.slice(1);
                return {
                    ...state,
                    snackBarOpen:true,
                    snackBarMsgInfo: newMsg,
                    snackBarMsgQueue: newMsgQueue,
                }
            }else{
                return state;
            }
        },
        [APPLICATION_SET_SNACKBAR_STATUS]:(state,action)=>{
            const option:IApplicationSetSnackbarStatusOption = action.payload as IApplicationSetSnackbarStatusOption;
            return {
                ...state,
                snackBarOpen:option.open,
            };
        },
        [APPLICATION_LAUNCH_BAR_TOGGLE_COLLAPSE]:(state,action)=>({
            ...state,
            launchBarCollapse:!state.launchBarCollapse,
        }),
        [APPLICATION_NETWORK_ONLINE]:(state,action)=>({
            ...state,
            offline:false,
        }),
        [APPLICATION_NETWORK_OFFLINE]:(state,action)=>({
            ...state,
            offline:true,
        }),
    } as any,initState);
}