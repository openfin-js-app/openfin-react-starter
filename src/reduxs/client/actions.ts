import { createAction, ActionFunctionAny, Action } from 'redux-actions';
import { IClientState } from './types';

export const CLIENT_SET_VALUE:string = 'CLIENT_SET_VALUE';
export const clientSetValue:ActionFunctionAny<Action<Partial<IClientState>>> = createAction(CLIENT_SET_VALUE,
    (option:Partial<IClientState>)=>(option)
);