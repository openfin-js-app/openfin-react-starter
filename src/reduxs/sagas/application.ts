import { buffers, delay } from 'redux-saga';
import { all, call, put, take, takeLatest, takeEvery, fork, select, actionChannel } from 'redux-saga/effects';
import { Docking ,System, Event, Window } from '@albertli90/redux-openfin';

import hist from '../../utils/history';

import { launchBarItems } from '../../layouts/LaunchBar/LaunchBarData';

import {
    APPLICATION_STARTED,
    APPLICATION_CHILD_STARTED,
    APPLICATION_NEW_SNACKBAR,
    APPLICATION_CLOSE_SNACKBAR,
    applicationUpdateDockStatus,
    APPLICATION_TOGGLE_WINDOW_STATE,
    applicationNewSnackbar,
    applicationReady,
    applicationSetSnackbarStatus,
    applicationProcessSnackbarQueue,
    APPLICATION_LAUNCH_BAR_TOGGLE,
    APPLICATION_LAUNCH_BAR_TOGGLE_COLLAPSE,
    APPLICATION_LAUNCH_NEW_WINDOW,
    configLoadFromDexie,
} from '..';

import { configUpdateNewWindowPosition } from '..';
import { GetGroupResPayload} from "@albertli90/redux-openfin/window/types";

const ENABLE_LOADING_VIEW=process.env.REACT_APP_ENABLE_LOADING_VIEW.toLowerCase() === 'true';

const LOADING_BANNER_WIDTH = parseInt(process.env.REACT_APP_LOADING_BANNER_WIDTH, 10);
const LOADING_BANNER_HEIGHT = parseInt(process.env.REACT_APP_LOADING_BANNER_HEIGHT, 10);
const DEFAULT_WIDTH = parseInt(process.env.REACT_APP_DEFAULT_APP_WIDTH, 10);
const DEFAULT_HEIGHT = parseInt(process.env.REACT_APP_DEFAULT_APP_HEIGHT, 10);

const previousBaseWindow={
    url:null,
    top:null,
    left:null,
    width:null,
    height:null,
};

export const getLaunchBarCollapse = state => state.application.launchBarCollapse;
export const getWindowsState = state => state.application.windowsState;
export const getNewWindowTop = state => state.config.application.newWinTop;
export const getNewWindowLeft = state => state.config.application.newWinLeft;
export const getNewWindowWidth = state => state.config.application.newWinWidth;
export const getNewWindowHeight = state => state.config.application.newWinHeight;

export function* handleRedirectToLoadingView(monitorRect) {

    const WINDOW_WIDTH  = monitorRect.right - monitorRect.left;
    const WINDOW_HEIGHT = monitorRect.bottom - monitorRect.top;
    const _LOADING_BANNER_WIDTH     = Math.min( LOADING_BANNER_WIDTH, WINDOW_WIDTH * 0.6387 );
    const _LOADING_BANNER_HEIGHT    = Math.min( LOADING_BANNER_HEIGHT, WINDOW_HEIGHT * 0.324074 );

    yield call(Window.asyncs.updateOptions,Window.actions.updateOptions({
        options:{resizable:false}
    }));

    yield call(Window.asyncs.setBounds,Window.actions.setBounds({
        left:(monitorRect.right - monitorRect.left)/2 - _LOADING_BANNER_WIDTH/2,
        top:(monitorRect.bottom - monitorRect.top)/2 - _LOADING_BANNER_HEIGHT/2,
        width:_LOADING_BANNER_WIDTH,
        height: _LOADING_BANNER_HEIGHT,
    }));
}

export function* handleRedirectFromLoadingView(monitorRect) {

    const WINDOW_WIDTH      = monitorRect.right - monitorRect.left;
    const WINDOW_HEIGHT     = monitorRect.bottom - monitorRect.top;
    const _DEFAULT_WIDTH    = Math.min( DEFAULT_WIDTH, WINDOW_WIDTH * 0.80 );
    const _DEFAULT_HEIGHT   = Math.min( DEFAULT_HEIGHT, WINDOW_HEIGHT * 0.648148 );

    // after the sagas loaded, redirect to default page/view
    if (process.env.REACT_APP_DEFAULT_VIEW_URL && process.env.REACT_APP_DEFAULT_VIEW_URL.length > 0){
        if (process.env.NODE_ENV !== 'test'){
            hist.push(process.env.REACT_APP_DEFAULT_VIEW_URL);
        }
        yield call(Window.asyncs.updateOptions,Window.actions.updateOptions({
            options:{
                resizable:true,
            }
        }));

        yield call(Window.asyncs.setBounds,Window.actions.setBounds({
            left:(monitorRect.right - monitorRect.left)/2 - _DEFAULT_WIDTH/2,
            top:(monitorRect.bottom - monitorRect.top)/2 - _DEFAULT_HEIGHT/2,
            width:_DEFAULT_WIDTH,
            height: _DEFAULT_HEIGHT,
        }));

    }else{
        // switch to launchBar
        const launchBarCollapse = yield select(getLaunchBarCollapse);

        yield call(Window.asyncs.updateOptions,Window.actions.updateOptions({
            options:{resizable:false}
        }));

        previousBaseWindow.url='/dashboard/view-one';
        previousBaseWindow.top=(monitorRect.bottom - monitorRect.top)/2 - _DEFAULT_HEIGHT/2;
        previousBaseWindow.left=(monitorRect.right - monitorRect.left)/2 - _DEFAULT_WIDTH/2;
        previousBaseWindow.width=_DEFAULT_WIDTH;
        previousBaseWindow.height=_DEFAULT_HEIGHT;

        if(launchBarCollapse){
            yield call(Window.asyncs.setBounds,Window.actions.setBounds({
                left:(monitorRect.right - monitorRect.left)/2,
                top:(monitorRect.bottom - monitorRect.top)/4,
                width:88,
                height:64,
            }));
        }else{
            yield call(Window.asyncs.setBounds,Window.actions.setBounds({
                left:(monitorRect.right - monitorRect.left)/2,
                top:(monitorRect.bottom - monitorRect.top)/4,
                width:launchBarItems.length<10?launchBarItems.length*64+88:664,
                height:64,
            }));
        }
        if (process.env.NODE_ENV !== 'test'){
            hist.push('/launchBar');
        }

    }
}

export function* handleApplicationLoading() {

    const currentIsLoadingView =
        (new URL(window.location.href).pathname.indexOf('loading')>-1) ||
        (new URL(window.location.href).pathname.indexOf('index.html')>-1);


    const monitorInfoAction = yield call(System.asyncs.getMonitorInfo,System.actions.getMonitorInfo({}));
    const monitorRect = monitorInfoAction.payload.primaryMonitor.monitorRect;

    yield call(Window.asyncs.setAsForeground,Window.actions.setAsForeground({}));

    if (ENABLE_LOADING_VIEW && currentIsLoadingView){
        yield* handleRedirectToLoadingView(monitorRect) as any;
    }

    yield all([
        put.resolve(configLoadFromDexie()),
        call(System.asyncs.getMachineId,System.actions.getMachineId({})),
        // getDeviceUserId might fail, thus use flux syntax........
        put.resolve(System.actions.getDeviceUserId({})),
        take(System.actions.GET_DEVICE_USER_ID_RES),
        call(System.asyncs.getEnvironmentVariable,System.actions.getEnvironmentVariable({env:'USERNAME'})),
        call(System.asyncs.getEnvironmentVariable,System.actions.getEnvironmentVariable({env:'computername'})),
        call(System.asyncs.getEnvironmentVariable,System.actions.getEnvironmentVariable({env:'HOSTNAME'})),
        call(System.asyncs.getVersion,System.actions.getVersion({})),
        call(System.asyncs.getHostSpecs,System.actions.getHostSpecs({})),
        call(Window.asyncs.getState,Window.actions.getState({})),
        // delay for loading view render, could be removed
        call(delay,5000),
    ]);

    yield put.resolve(applicationReady());

    if (ENABLE_LOADING_VIEW && currentIsLoadingView){
        yield* handleRedirectFromLoadingView(monitorRect) as any;
    }
}

export function* handleApplicationChildLoading() {
    yield all([
        call(Window.asyncs.getBounds,Window.actions.getBounds({})),
        put.resolve(configLoadFromDexie()),
    ]);

    const groupedWindowsRes = yield call(Window.asyncs.getGroup,Window.actions.getGroup({}));
    const groupedWindows = (groupedWindowsRes.payload as GetGroupResPayload).windows;
    if(groupedWindows && groupedWindows.length > 0){
        yield put(applicationUpdateDockStatus(true));
    }else{
        yield put(applicationUpdateDockStatus(false));
    }

    yield put.resolve(applicationReady());
}

export function* handleApplicationExit() {
    // -------------------------------start of app codes -----------------------------------------------

    // do something cleaning up before shutdonw~

    // ---------------------------------end of app codes -----------------------------------------------


    yield put.resolve(Window.actions.close({force:true}));
}

export function* handleApplicationAddNewSnackBar() {
    const state = yield select();
    if(state.snackBarOpen){
        yield put(applicationSetSnackbarStatus({open:false}));
    }else{
        yield put(applicationProcessSnackbarQueue());
    }
}

export function* handleApplicationCloseSnackBar(action) {
    const {event, reason} = action.payload;
    if(reason!=='clickaway'){
        return;
    }else{
        yield put(applicationSetSnackbarStatus({open:false}));
    }
}

export function* handleToggleWindowState(){
    const windowState = yield select(getWindowsState);
    if (windowState === 'maximized'){
        yield call(Window.asyncs.restore,Window.actions.restore({}));
    }else if (windowState === 'normal'){
        yield call(Window.asyncs.maximize,Window.actions.maximize({}));
    }
}

export function* handleApplicationLaunchBarToggle(){

    const launchBarCollapse = yield select(getLaunchBarCollapse);
    const getBoundsAction = yield call(Window.asyncs.getBounds,Window.actions.getBounds({}));
    const getBoundsActionPayload = getBoundsAction.payload;

    if (window.location.href.toLowerCase().endsWith('launchbar')){
        // switch to main panel
        yield call(Window.asyncs.updateOptions,Window.actions.updateOptions({
            options:{
                resizable:true,
            }
        }));
        yield call(Window.asyncs.setBounds,Window.actions.setBounds({
            left:previousBaseWindow.left?previousBaseWindow.left:parseInt(process.env.REACT_APP_NEW_WINDOW_LEFT,10),
            top:previousBaseWindow.top?previousBaseWindow.top:parseInt(process.env.REACT_APP_NEW_WINDOW_TOP,10),
            width:previousBaseWindow.width?previousBaseWindow.width:parseInt(process.env.REACT_APP_NEW_WINDOW_WIDTH,10),
            height:previousBaseWindow.height?previousBaseWindow.height:parseInt(process.env.REACT_APP_NEW_WINDOW_HEIGHT,10),
        }));
        if (process.env.NODE_ENV !== 'test'){
            hist.push(previousBaseWindow.url?previousBaseWindow.url:process.env.REACT_APP_DEFAULT_DASHBOARD_VIEW_URL);
        }
    }else{
        // switch to launchBar
        yield call(Window.asyncs.updateOptions,Window.actions.updateOptions({
            options:{
                resizable:false,
            }
        }));
        previousBaseWindow.url = (new URL(window.location.href)).pathname;
        previousBaseWindow.top = getBoundsActionPayload.top;
        previousBaseWindow.left = getBoundsActionPayload.left;
        previousBaseWindow.width = getBoundsActionPayload.width;
        previousBaseWindow.height = getBoundsActionPayload.height;

        if (launchBarCollapse){
            yield call(Window.asyncs.setBounds,Window.actions.setBounds({
                left:getBoundsActionPayload.left,
                top:getBoundsActionPayload.top,
                width:88,
                height:64,
            }));
        }else{
            yield call(Window.asyncs.setBounds,Window.actions.setBounds({
                left:getBoundsActionPayload.left,
                top:getBoundsActionPayload.top,
                width:launchBarItems.length<10? launchBarItems.length*64+88:664,
                height:64,
            }));
        }
        if (process.env.NODE_ENV !== 'test'){
            hist.push('/launchBar');
        }

    }
}

export function* handleApplicationLaunchBarToggleCollapse() {
    const launchBarCollapse = yield select(getLaunchBarCollapse);

    const getBoundsAction = yield call(Window.asyncs.getBounds,Window.actions.getBounds({}));
    const getBoundsActionPayload = getBoundsAction.payload;

    if (launchBarCollapse){
        yield call(Window.asyncs.setBounds,Window.actions.setBounds({
            left:getBoundsActionPayload.left,
            top:getBoundsActionPayload.top,
            width:88,
            height:64,
        }));
    }else{
        yield call(Window.asyncs.setBounds,Window.actions.setBounds({
            left:getBoundsActionPayload.left,
            top:getBoundsActionPayload.top,
            width:launchBarItems.length<10? launchBarItems.length*64+88:664,
            height:64,
        }));
    }

}

export function* handleApplicationLaunchNewWindow(action) {
    const appJson = action.payload;
    const defaultWidth = yield select(getNewWindowWidth);
    const defaultHeight = yield select(getNewWindowHeight);
    const defaultTop = yield select(getNewWindowTop);
    const defaultLeft = yield select(getNewWindowLeft);

    if(!appJson.defaultWidth){ appJson.defaultWidth = defaultWidth}
    if(!appJson.defaultHeight){ appJson.defaultHeight = defaultHeight}
    if(!appJson.defaultTop){ appJson.defaultTop = defaultTop}
    if(!appJson.defaultLeft){ appJson.defaultLeft = defaultLeft}

    yield call(Window.asyncs.newWindow,Window.actions.newWindow(appJson));

    yield put(configUpdateNewWindowPosition());

}

export function* handleGroupChanged(action){
    const {
        sourceWindowName, targetWindowName, memeberOf, reason
    } = action.payload;


    if (reason === Docking.types.GroupEventReason.JOIN){
        if(sourceWindowName === window.name){
            yield put(applicationNewSnackbar({
                message:'Joined group',
                variant:'primary'
            }))
        }else if (targetWindowName === window.name){
            yield put(applicationNewSnackbar({
                message:'Been joined',
                variant:'rose'
            }))
        }
    }else if (
        reason === Docking.types.GroupEventReason.LEAVE &&
        sourceWindowName === window.name
    ){
        yield put(applicationNewSnackbar({
            message:'Left group',
            variant:'primary'
        }))
    }else if (
        reason === Docking.types.GroupEventReason.DISBAND &&
        sourceWindowName === window.name
    ){
        yield put(applicationNewSnackbar({
            message:'Got disbanded',
            variant:'rose'
        }))
    }

}

export default function* (){
    yield takeLatest(APPLICATION_STARTED,handleApplicationLoading);
    yield takeLatest(APPLICATION_CHILD_STARTED,handleApplicationChildLoading);
    yield takeLatest(Event.actionDicts.windowEventDictByName['close-requested'].type,handleApplicationExit);
    yield takeLatest(APPLICATION_TOGGLE_WINDOW_STATE,handleToggleWindowState);
    yield takeLatest(APPLICATION_NEW_SNACKBAR,handleApplicationAddNewSnackBar);
    yield takeLatest(APPLICATION_CLOSE_SNACKBAR,handleApplicationCloseSnackBar);
    yield takeLatest(APPLICATION_LAUNCH_BAR_TOGGLE,handleApplicationLaunchBarToggle);
    yield takeLatest(APPLICATION_LAUNCH_BAR_TOGGLE_COLLAPSE,handleApplicationLaunchBarToggleCollapse);
    yield takeLatest(APPLICATION_LAUNCH_NEW_WINDOW,handleApplicationLaunchNewWindow);
    yield takeEvery(Event.actionDicts.windowEventDictByName['group-changed'].type,handleGroupChanged);
}