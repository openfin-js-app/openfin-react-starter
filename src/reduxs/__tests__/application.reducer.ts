import { System, Window } from '@albertli90/redux-openfin';
import * as Actions from '../application/actions';
import reducerCreater from '../application/reducer';

const reducer = reducerCreater();

describe('Application reducer',()=>{

    it('System.actions.GET_MACHINE_ID_RES reduced correctly',()=>{
        const action:any = {type:System.actions.GET_MACHINE_ID_RES, payload:{id:'id'}};
        expect(reducer(undefined,action)).toMatchSnapshot();
    });


    it('System.actions.GET_DEVICE_USER_ID_RES reduced correctly',()=>{
        const action:any = {type:System.actions.GET_DEVICE_USER_ID_RES, payload:{id:'id'}};
        expect(reducer(undefined,action)).toMatchSnapshot();
    });


    it('System.actions.GET_ENVIRONMENT_VARIABLE_RES reduced correctly - username',()=>{
        const action:any = {type:System.actions.GET_ENVIRONMENT_VARIABLE_RES, payload:{
            env:'username', value:'usernameVal'
            }};
        expect(reducer(undefined,action)).toMatchSnapshot();
    });

    it('System.actions.GET_ENVIRONMENT_VARIABLE_RES reduced correctly - computername',()=>{
        const action:any = {type:System.actions.GET_ENVIRONMENT_VARIABLE_RES, payload:{
                env:'computername', value:'computernameVal'
            }};
        expect(reducer(undefined,action)).toMatchSnapshot();
    });

    it('System.actions.GET_ENVIRONMENT_VARIABLE_RES reduced correctly - unmatched',()=>{
        const action:any = {type:System.actions.GET_ENVIRONMENT_VARIABLE_RES, payload:{
                env:'someEnv', value:'someVal'
            }};
        expect(reducer(undefined,action)).toMatchSnapshot();
    });

    it('System.actions.GET_VERSION_RES reduced correctly',()=>{
        const action:any = {type:System.actions.GET_VERSION_RES, payload:{
            version:'versionStr'
            }};
        expect(reducer(undefined,action)).toMatchSnapshot();
    });

    it('System.actions.GET_HOST_SPECS_RES reduced correctly',()=>{
        const action:any = {type:System.actions.GET_HOST_SPECS_RES, payload:{
                type: 'GET_HOST_SPECS_RES'
            }};
        expect(reducer(undefined,action)).toMatchSnapshot();
    });

    it('Window.actions.GET_STATE_RES reduced correctly',()=>{
        const action:any = {type:Window.actions.GET_STATE_RES, payload:{
                state:'windowState'
            }};
        expect(reducer(undefined,action)).toMatchSnapshot();
    });

    it('APPLICATION_READY reduced correctly',()=>{
        const action:any = Actions.applicationReady();
        expect(reducer(undefined,action)).toMatchSnapshot();
    });

    it('APPLICATION_DRAWER_TOGGLE reduced correctly',()=>{
        const action:any = Actions.applicationDrawerToggle();
        expect(reducer(undefined,action)).toMatchSnapshot();
    });

    it('APPLICATION_NEW_SNACKBAR reduced correctly',()=>{
        const action:any = Actions.applicationNewSnackbar({
            message:'Snackbar message',
            variant:'primary',
        });
        const resultState = reducer(undefined,action);
        delete resultState.snackBarMsgQueue[0].key;
        expect(resultState).toMatchSnapshot();
    });

    it('APPLICATION_PROCESS_SNACKBAR_QUEUE reduced correctly - nonempty',()=>{
        const action:any = Actions.applicationProcessSnackbarQueue();
        expect(reducer({
            snackBarOpen:false,
            snackBarMsgInfo:{},
            snackBarMsgQueue:[{
                message:'Snackbar message',
                variant:'primary',
            }]
        },action)).toMatchSnapshot();
    });

    it('APPLICATION_PROCESS_SNACKBAR_QUEUE reduced correctly - empty',()=>{
        const action:any = Actions.applicationProcessSnackbarQueue();
        expect(reducer(undefined,action)).toMatchSnapshot();
    });

    it('APPLICATION_SET_SNACKBAR_STATUS reduced correctly - empty',()=>{
        const action:any = Actions.applicationSetSnackbarStatus({
            open:true,
        });
        expect(reducer(undefined,action)).toMatchSnapshot();
    });

    it('APPLICATION_LAUNCH_BAR_TOGGLE_COLLAPSE reduced correctly - empty',()=>{
        const action:any = Actions.applicationLaunchBarToggleCollapse();
        expect(reducer(undefined,action)).toMatchSnapshot();
    });

});