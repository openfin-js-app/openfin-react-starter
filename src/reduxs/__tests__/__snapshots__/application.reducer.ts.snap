// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Application reducer APPLICATION_DRAWER_TOGGLE reduced correctly 1`] = `
Object {
  "computerName": "",
  "deviceUserId": null,
  "drawerOpen": false,
  "launchBarCollapse": false,
  "loading": true,
  "machineId": null,
  "openfinHostSpec": Object {},
  "openfinVersion": "n/a",
  "snackBarMsgInfo": Object {},
  "snackBarMsgQueue": Array [],
  "snackBarOpen": false,
  "username": "",
  "windowsState": "normal",
}
`;

exports[`Application reducer APPLICATION_LAUNCH_BAR_TOGGLE_COLLAPSE reduced correctly - empty 1`] = `
Object {
  "computerName": "",
  "deviceUserId": null,
  "drawerOpen": true,
  "launchBarCollapse": true,
  "loading": true,
  "machineId": null,
  "openfinHostSpec": Object {},
  "openfinVersion": "n/a",
  "snackBarMsgInfo": Object {},
  "snackBarMsgQueue": Array [],
  "snackBarOpen": false,
  "username": "",
  "windowsState": "normal",
}
`;

exports[`Application reducer APPLICATION_NEW_SNACKBAR reduced correctly 1`] = `
Object {
  "computerName": "",
  "deviceUserId": null,
  "drawerOpen": true,
  "launchBarCollapse": false,
  "loading": true,
  "machineId": null,
  "openfinHostSpec": Object {},
  "openfinVersion": "n/a",
  "snackBarMsgInfo": Object {},
  "snackBarMsgQueue": Array [
    Object {
      "message": "Snackbar message",
      "variant": "primary",
    },
  ],
  "snackBarOpen": false,
  "username": "",
  "windowsState": "normal",
}
`;

exports[`Application reducer APPLICATION_PROCESS_SNACKBAR_QUEUE reduced correctly - empty 1`] = `
Object {
  "computerName": "",
  "deviceUserId": null,
  "drawerOpen": true,
  "launchBarCollapse": false,
  "loading": true,
  "machineId": null,
  "openfinHostSpec": Object {},
  "openfinVersion": "n/a",
  "snackBarMsgInfo": Object {},
  "snackBarMsgQueue": Array [],
  "snackBarOpen": false,
  "username": "",
  "windowsState": "normal",
}
`;

exports[`Application reducer APPLICATION_PROCESS_SNACKBAR_QUEUE reduced correctly - nonempty 1`] = `
Object {
  "snackBarMsgInfo": Object {
    "message": "Snackbar message",
    "variant": "primary",
  },
  "snackBarMsgQueue": Array [],
  "snackBarOpen": true,
}
`;

exports[`Application reducer APPLICATION_READY reduced correctly 1`] = `
Object {
  "computerName": "",
  "deviceUserId": null,
  "drawerOpen": true,
  "launchBarCollapse": false,
  "loading": false,
  "machineId": null,
  "openfinHostSpec": Object {},
  "openfinVersion": "n/a",
  "snackBarMsgInfo": Object {},
  "snackBarMsgQueue": Array [],
  "snackBarOpen": false,
  "username": "",
  "windowsState": "normal",
}
`;

exports[`Application reducer APPLICATION_SET_SNACKBAR_STATUS reduced correctly - empty 1`] = `
Object {
  "computerName": "",
  "deviceUserId": null,
  "drawerOpen": true,
  "launchBarCollapse": false,
  "loading": true,
  "machineId": null,
  "openfinHostSpec": Object {},
  "openfinVersion": "n/a",
  "snackBarMsgInfo": Object {},
  "snackBarMsgQueue": Array [],
  "snackBarOpen": true,
  "username": "",
  "windowsState": "normal",
}
`;

exports[`Application reducer System.actions.GET_DEVICE_USER_ID_RES reduced correctly 1`] = `
Object {
  "computerName": "",
  "deviceUserId": "id",
  "drawerOpen": true,
  "launchBarCollapse": false,
  "loading": true,
  "machineId": null,
  "openfinHostSpec": Object {},
  "openfinVersion": "n/a",
  "snackBarMsgInfo": Object {},
  "snackBarMsgQueue": Array [],
  "snackBarOpen": false,
  "username": "",
  "windowsState": "normal",
}
`;

exports[`Application reducer System.actions.GET_ENVIRONMENT_VARIABLE_RES reduced correctly - computername 1`] = `
Object {
  "computerName": "computernameVal",
  "deviceUserId": null,
  "drawerOpen": true,
  "launchBarCollapse": false,
  "loading": true,
  "machineId": null,
  "openfinHostSpec": Object {},
  "openfinVersion": "n/a",
  "snackBarMsgInfo": Object {},
  "snackBarMsgQueue": Array [],
  "snackBarOpen": false,
  "username": "",
  "windowsState": "normal",
}
`;

exports[`Application reducer System.actions.GET_ENVIRONMENT_VARIABLE_RES reduced correctly - unmatched 1`] = `
Object {
  "computerName": "",
  "deviceUserId": null,
  "drawerOpen": true,
  "launchBarCollapse": false,
  "loading": true,
  "machineId": null,
  "openfinHostSpec": Object {},
  "openfinVersion": "n/a",
  "snackBarMsgInfo": Object {},
  "snackBarMsgQueue": Array [],
  "snackBarOpen": false,
  "username": "",
  "windowsState": "normal",
}
`;

exports[`Application reducer System.actions.GET_ENVIRONMENT_VARIABLE_RES reduced correctly - username 1`] = `
Object {
  "computerName": "",
  "deviceUserId": null,
  "drawerOpen": true,
  "launchBarCollapse": false,
  "loading": true,
  "machineId": null,
  "openfinHostSpec": Object {},
  "openfinVersion": "n/a",
  "snackBarMsgInfo": Object {},
  "snackBarMsgQueue": Array [],
  "snackBarOpen": false,
  "username": "usernameVal",
  "windowsState": "normal",
}
`;

exports[`Application reducer System.actions.GET_HOST_SPECS_RES reduced correctly 1`] = `
Object {
  "computerName": "",
  "deviceUserId": null,
  "drawerOpen": true,
  "launchBarCollapse": false,
  "loading": true,
  "machineId": null,
  "openfinHostSpec": Object {
    "type": "GET_HOST_SPECS_RES",
  },
  "openfinVersion": "n/a",
  "snackBarMsgInfo": Object {},
  "snackBarMsgQueue": Array [],
  "snackBarOpen": false,
  "username": "",
  "windowsState": "normal",
}
`;

exports[`Application reducer System.actions.GET_MACHINE_ID_RES reduced correctly 1`] = `
Object {
  "computerName": "",
  "deviceUserId": null,
  "drawerOpen": true,
  "launchBarCollapse": false,
  "loading": true,
  "machineId": "id",
  "openfinHostSpec": Object {},
  "openfinVersion": "n/a",
  "snackBarMsgInfo": Object {},
  "snackBarMsgQueue": Array [],
  "snackBarOpen": false,
  "username": "",
  "windowsState": "normal",
}
`;

exports[`Application reducer System.actions.GET_VERSION_RES reduced correctly 1`] = `
Object {
  "computerName": "",
  "deviceUserId": null,
  "drawerOpen": true,
  "launchBarCollapse": false,
  "loading": true,
  "machineId": null,
  "openfinHostSpec": Object {},
  "openfinVersion": "versionStr",
  "snackBarMsgInfo": Object {},
  "snackBarMsgQueue": Array [],
  "snackBarOpen": false,
  "username": "",
  "windowsState": "normal",
}
`;

exports[`Application reducer Window.actions.GET_STATE_RES reduced correctly 1`] = `
Object {
  "computerName": "",
  "deviceUserId": null,
  "drawerOpen": true,
  "launchBarCollapse": false,
  "loading": true,
  "machineId": null,
  "openfinHostSpec": Object {},
  "openfinVersion": "n/a",
  "snackBarMsgInfo": Object {},
  "snackBarMsgQueue": Array [],
  "snackBarOpen": false,
  "username": "",
  "windowsState": "windowState",
}
`;
